Python/parking_lot.c

```diff
@@ -102,7 +102,13 @@ _PySemaphore_PlatformWait(_PySemaphore *sema, PyTime_t timeout)
         millis = INFINITE;
     }
     else {
-        millis = (DWORD) (timeout / 1000000);
+        // Prevent overflow with clamping the result
+        if ((PyTime_t)PY_DWORD_MAX * 1000000 < timeout) {
```

Comment:
IIUC Victor's original comment, https://github.com/python/cpython/pull/112733#pullrequestreview-1766327004
He suggested to use `_PyTime_As100Nanoseconds`, but maybe it should be `_PyTime_AsMilliseconds` because we convert nano_seconds to milliseconds.

But either we use  `_PyTime_As100Nanoseconds` or  `_PyTime_AsMilliseconds` it does not prevent overflow from PyTime_t to DWORD, because it's about rounding logic between PyTime_t and PyTime_t not between PyTime_t to DWORD.

Please let me know if I miss something.